iothub/device/src/ClientSettings/ClientOptions.cs

```diff
@@ -10,10 +10,16 @@ namespace Microsoft.Azure.Devices.Client
     /// </summary>
     public class ClientOptions
     {
+        /// <summary>
+        /// The transport type to use (i.e., AMQP, MQTT, HTTP), including whether to use TCP or web sockets where applicable.
+        /// </summary>
+        public TransportType TransportType { get; set; } = TransportType.Amqp;
```

Comment:
How are we accepting the transport settings? Is that also going to be a part of client options?

Reply:
Yes, but a next step in this series of PRs.

Reply:
Also, I plan to only accept a single one, not an array.

Thoughts?

Reply:
That sounds good to me. +1 to removing the fallback code.
We'd need a base class for the transport settings then, if we don't have one already.